File: Data Structure/Lab 9/CopyList.c
//WAP to copy a linked list.

#include<stdio.h>
#include<stdlib.h>

struct Node
{
    int info;
    struct Node * link;
};

struct Node *first1;
struct Node *first2;

void insertAtFirst(int x){

    struct Node *newNode = (struct Node *)malloc(sizeof(struct Node*));

    newNode->info = x;
    newNode->link = first1;

    first1 = newNode;
}

void display(struct Node *first){
    struct Node *save = first;

    if(first == NULL){
        printf("List is Empty!");
        return;
    }

    while(save != NULL){
        printf("%d->",save->info);
        save = save->link;
    }
    printf("NULL");
}

void insertAtLast(int x){
    struct Node *newNode = NULL;
    struct Node *save = first2;

    while(save->link != NULL){
        save = save->link;
    }
    save->link = newNode;
    newNode->link = NULL;
}

void CopyList(struct Node *first){
    struct Node *save1 = first;

    while (save1 != NULL)
    {
        insertAtLast(save1 -> info);
        save1 = save1->link;
    }
    
}

void main(){
    insertAtFirst(1);
    insertAtFirst(2);
    insertAtFirst(3);
    insertAtFirst(4);

    CopyList(first1);
    display(first1);

}

File: Data Structure(Java)/Lab 7/CopyList.java
//WAP to copy a linked list.

class Node{
    int info;
    Node link;

    Node(int info){
        this.info = info;
        this.link = null;
    }
}

public class CopyList {

    static Node first1 = null;
    static Node first2 = null;

    static void insertAtFirst(int x){
        Node newNode = new Node(x);

        newNode.info = x;
        newNode.link = first1;

        first1 = newNode;

    }

    static void displayNodes(Node first){

        if(first == null){
            System.out.println("List is empty!!");
            return;
        }

        Node save = first;

        while(save != null){
            System.out.print(save.info+"->");
            save = save.link;
        }
        System.out.println("null");
    }

    static void insertAtlast(int x){
        Node newNode = new Node(x);

        if(first2 == null){
            first2 = newNode;
            return;
        }

        Node save = first2;

        while(save.link != null){
            save = save.link;
        }
        save.link = newNode;
    }

    static void Copylist(){
        Node save = first1;

        while(save != null){
            insertAtlast(save.info);
            save = save.link;
        }
    }

    public static void main(String[] args) {
        
        insertAtFirst(9);
        insertAtFirst(8);
        insertAtFirst(7);
        insertAtFirst(6);

        displayNodes(first1);

        Copylist();

        displayNodes(first2);
    }
}